File: python/day4.py
#!/usr/local/bin/python3

import hashlib

md5_key = 'ckczppom'
for prefix in ['0' * 5, '0' * 6]:
    num = 1
    while True:
        string = md5_key + str(num)
        result = hashlib.md5(string.encode()).hexdigest()
        if result.startswith(prefix):
            print(num, result)
            break
        num += 1

File: 2015/javascript/day4.js
#!/usr/local/bin/node

'use strict';

var crypto = require('crypto')


var md5_digest = function (string) {
    var hash = crypto.createHash('md5');
    hash.update(string, 'utf-8');
    return hash.digest('hex');
};

var md5_prefix = 'ckczppom';
var num = 1;
['00000', '000000'].forEach(function (prefix) {
   while (true) {
       var digest = md5_digest(md5_prefix + String(num));
       if (digest.startsWith(prefix)) {
           console.log(num, digest);
           break;
       } else {
           num += 1;
       }
   }
});
